tests/queries/0_stateless/02983_variant_data_types.sh

```diff
@@ -0,0 +1,312 @@
+    run_select "UInt16"
+    drop_table
+
+    create_table "$1" "UInt32"
+    $CH_CLIENT -q  "insert into test select number, (number % 2 ? 'str_' || toString(number) : number::UInt32) from numbers(3);"
+    run_select "UInt32"
+    drop_table
+
+    create_table "$1" "UInt64"
+    $CH_CLIENT -q  "insert into test select number, (number % 2 ? 'str_' || toString(number) : number) from numbers(3);"
+    run_select "UInt64"
+    drop_table
+    
+    create_table "$1" "UInt128"
+    $CH_CLIENT -q  "insert into test select number, (number % 2 ? 'str_' || toString(number) : number::UInt128) from numbers(3);"
+    run_select "UInt128"
+    drop_table
+
+    create_table "$1" "UInt256"
+    $CH_CLIENT -q  "insert into test select number, (number % 2 ? 'str_' || toString(number) : number::UInt256) from numbers(3);"
+    run_select "UInt256"
+    drop_table
+
+    create_table "$1" "Int8"
+    $CH_CLIENT -q  "insert into test select number, (number % 2 ? 'str_' || toString(number) : number::Int8) from numbers(3);"
+    run_select "Int8"
+    drop_table
+    
+    create_table "$1" "Int16"
+    $CH_CLIENT -q  "insert into test select number, (number % 2 ? 'str_' || toString(number) : number::Int16) from numbers(3);"
+    run_select "Int16"
+    drop_table
+    
+    create_table "$1" "Int32"
+    $CH_CLIENT -q  "insert into test select number, (number % 2 ? 'str_' || toString(number) : number::Int32) from numbers(3);"
+    run_select "Int32"
+    drop_table
+    
+    create_table "$1" "Int64"
+    $CH_CLIENT -q  "insert into test select number, (number % 2 ? 'str_' || toString(number) : number::Int64) from numbers(3);"
+    run_select "Int64"
+    drop_table
+    
+    create_table "$1" "Int128"
+    $CH_CLIENT -q  "insert into test select number, (number % 2 ? 'str_' || toString(number) : number::Int128) from numbers(3);"
+    run_select "Int128"
+    drop_table
+    
+    create_table "$1" "Int256"
+    $CH_CLIENT -q  "insert into test select number, (number % 2 ? 'str_' || toString(number) : number::Int256) from numbers(3);"
+    run_select "Int256"
+    drop_table
+}
+
+function test_floating_point_numbers()
+{
+    echo "Floating Point Types"
+
+    create_table "$1" "Float32"
+    $CH_CLIENT -q  "insert into test select 1, round(1 - 0.9, 4)::Float32 FROM system.numbers LIMIT 3;"
+    run_select "Float32"
+    drop_table
+
+    create_table "$1" "Float64"
+    $CH_CLIENT -q  "insert into test select 1, (1 - 0.9)::Float64 FROM system.numbers LIMIT 3;"
+    run_select "Float64"
+    drop_table
+    
+    #create_table "$1" "Decimal32"
+    #$CH_CLIENT -q  "insert into test select 1, toDecimal32(round(randCanonical(), 3)::Decimal32 , 3) FROM system.numbers LIMIT 3;"
+    #run_select "Decimal32"
+    #drop_table
+}
+
+function test_boolean()
+{
+    echo "Boolean Types"
+
+    create_table "$1" "Bool"   # create table works for Boolean but select only works for .Bool
```

Comment:
It's expected, for subcolumns we use only canocical type names. But definetely we need to state it in the docs, I will add it.